pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -723,6 +1001,17 @@ func handleModify(ctxArg interface{}, key string,
 
 	status.StartTime = ctx.delayBaseTime.Add(config.Delay)
 
+	updateSnapshotsInAIStatus(status, config)
+	if len(status.SnapStatus.SnapshotsToBeDeleted) > 0 {
+		// Mark the VM to be rebooted
```

Comment:
Does this mean that an app instance can be rebooted because the controller merely requested that some snapshot be deleted? For a user that would be a big (and nagative) surprise. Can we not defer such work until the next time the application is updated and/or rebooted?

Reply:
We should describe this in a way where it can be documented. Maybe down the road we can improve on this but the documentation is what we need now.

Reply:
I can test the approach with a deferred deletion. In theory, it's not that hard.

Reply:
I've tested it, looks fine.

Reply:
Fixed